examples/grpc/src/main.rs

```diff
@@ -111,68 +110,87 @@ impl Greeter for MyGreeter {
 }
 
 mod connector {
+    use hyper::Uri;
+    use hyper_util::client::legacy::connect::Connected;
+    use hyper_util::rt::TokioIo;
+    use std::task::{Context, Poll};
     use std::{future::Future, pin::Pin};
-
-    use hyper::{
-        client::connect::{Connected, Connection},
-        Uri,
-    };
-    use tokio::io::{AsyncRead, AsyncWrite};
+    use tokio::io;
+    use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+    use tonic::transport::server::TcpConnectInfo;
     use tower::Service;
     use turmoil::net::TcpStream;
 
-    type Fut = Pin<Box<dyn Future<Output = Result<TurmoilConnection, std::io::Error>> + Send>>;
+    #[derive(Clone)]
+    pub struct TurmoilTcpConnector;
+
+    impl Service<Uri> for TurmoilTcpConnector {
+        type Response = TokioIo<TurmoilTcpStream>;
+        type Error = io::Error;
+        type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send>>;
 
-    pub fn connector(
-    ) -> impl Service<Uri, Response = TurmoilConnection, Error = std::io::Error, Future = Fut> + Clone
-    {
-        tower::service_fn(|uri: Uri| {
+        fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+            Poll::Ready(Ok(()))
+        }
+
+        fn call(&mut self, uri: Uri) -> Self::Future {
             Box::pin(async move {
-                let conn = TcpStream::connect(uri.authority().unwrap().as_str()).await?;
-                Ok::<_, std::io::Error>(TurmoilConnection(conn))
-            }) as Fut
-        })
+                let stream = TcpStream::connect(uri.authority().unwrap().as_str()).await?;
+                Ok(TokioIo::new(TurmoilTcpStream(stream)))
+            })
+        }
     }
 
-    pub struct TurmoilConnection(turmoil::net::TcpStream);
+    pub struct TurmoilTcpStream(pub TcpStream);
 
-    impl AsyncRead for TurmoilConnection {
+    impl hyper_util::client::legacy::connect::Connection for TurmoilTcpStream {
+        fn connected(&self) -> Connected {
+            Connected::new()
+        }
+    }
+
+    impl tonic::transport::server::Connected for TurmoilTcpStream {
```

Comment:
It seems that `TokioIo` brings us pretty close to not having to implement all these traits for turmoil types, other than this one.

Reply:
I think we still need the `AsyncRead` / `AsyncWrite` implementations, or implementations for `hyper::rt::Read/Write` as TokioIo<T> requires T to impl either of those